### ISSUE TITLE
highlight capture groups with different color in str_view

### ISSUE THREAD

**Issue by dracodoc** [hadley/stringr#121]
If the regex expression have capture groups, can `str_view` highlight them in different colors? 

For example, my expression `"[^\\w](\\w*)$"` have some parts and a capture group, but I can't see which part is the capture group match inside the whole match in `str_view`.

**Comment by hadley** [comment]
I'd review a PR that implemented that, but I think it's currently outside the scope of `str_view`

**Comment by dracodoc** [comment]
I did spend quite some time trying to do this. It- seemed to be plausible, and I learned a lot from your code. 

However I found there are only 3 functions in stringr are capture groups awared, and they are also more about the matched value, not the match position in string. So I need to make a lot of efforts to get that position, maybe with duplicated regex seaches. 

Now reflect back, I think this may not be an important or often used feature. There are some software or websites providing regex match visualization, so user may check those for more specific needs.
